client/landing/jetpack-cloud/components/date-picker/index.jsx

```diff
@@ -18,80 +19,100 @@ import Gridicon from 'components/gridicon';
  */
 import './style.scss';
 
-const DATE_FORMAT = 'YYYY-MM-DD';
-
 class DatePicker extends Component {
 	static propTypes = {
 		siteId: PropTypes.number.isRequired,
-		selectedDateString: PropTypes.string.isRequired,
-		onChange: PropTypes.func.isRequired,
+		selectedDate: PropTypes.instanceOf( Date ).isRequired,
+		onDateChange: PropTypes.func.isRequired,
+		onDateRangeSelection: PropTypes.func.isRequired,
 	};
 
-	getFormattedDate = dateString => this.props.moment.parseZone( dateString ).format( DATE_FORMAT );
+	getDisplayDate = ( date, showTodayYesterday = true ) => {
+		const { moment, translate } = this.props;
+
+		const daysDiff = moment().diff( date, 'days' );
+		const yearToday = moment().format( 'YYYY' );
+		const yearDate = moment( date ).format( 'YYYY' );
+
+		let dateFormat = 'MMM D';
```

Comment:
let's collapse these lines via a ternary, unless there are other possible cases
```suggestion
		const dateFormat = yearToday === yearDate ? 'MMM D' : 'MMM D, YYYY';
```

Reply:
Also we might need to replace `MMM D, YYYY` with `LL`, since the order of the date is locale-specific

Reply:
Great! Improved it in the next PR: https://github.com/Automattic/wp-calypso/pull/40189

Reply:
> Also we might need to replace `MMM D, YYYY` with `LL`, since the order of the date is locale-specific

We have to discuss it with the design team. In our case, if we use the locale format in some localization could break completely the design (mobile), and for the case of the format `MMM D` we don't have any locale-specific in Moment.

**Example of the lengths:**
moment().format('ll');   // Mar 17, 2020 (USA)
moment().format('ll');   // 17 de mar. de 2020  (Spanish)

See how my-sites/stats does it, for short spaces, they use `MMM D` format.